### PR TITLE
Experimental windows provisioning (tested on windows 8)

### PR DESCRIPTION
**Pull request by kamino**
Known bug: probe name cannot be more than 15 characters

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -28,10 +25,14 @@ Vagrant.configure("2") do |config|
       config.vbguest.no_install = true
     end
   end
-
+
```
Comment (guillaumededrie) [inline Vagrantfile]: hum…

File: Vagrantfile [hunk 2]
```diff
@@ -28,10 +25,14 @@ Vagrant.configure("2") do |config|
       config.vbguest.no_install = true
     end
   end
-
+ 
   # Multi machine environment
   servers.each do |server|
     config.vm.define server['name'] do |machine|
+      machine.ssh.forward_agent = true
```
Comment (guillaumededrie) [inline Vagrantfile]: for what reason did you move these two lines?

File: ansible.cfg [hunk 3]
```diff
@@ -1,9 +1,14 @@
 [defaults]
-hash_behaviour = merge
-ansible_managed = Please do not change this file directly since it is managed by Ansible and will be overwritten
+hash_behaviour     = merge
```
Comment (guillaumededrie) [inline ansible.cfg]: argh :(

Comment (guillaumededrie) [inline ansible.cfg]: Please, don't let your editor dictate choices :(

Comment (kamino) [inline ansible.cfg]: GJ, I haven't seen that :+1:

File: Vagrantfile (hunk 1 above)
Comment (guillaumededrie) [inline Vagrantfile]: Be careful with extra-whitespaces, it's adding a change history into git.

File: Vagrantfile (hunk 2 above)
Comment (kamino) [inline Vagrantfile]: Simple artefact due to integration.

File: ansible.cfg (hunk 3 above)
Comment (guillaumededrie) [inline ansible.cfg]: Is it possible to remove these changes, as it's editor related and don't add information to the PR.

File: README.rst [hunk 4]
```diff
@@ -78,6 +77,40 @@ some useful commands with vagrant:
 
 for advanced usage of vagrant be sure to check `extras <docs/install_extras.rst>`_
 
+Experimental support for Windows provisioning
```
Comment (guillaumededrie) [inline README.rst]: I think we should generate our own box and put in on vagrant cloud Quarkslab user, even if we decide to not update it.

Comment (guillaumededrie) [inline README.rst]: As we did with the boxcutter/debian Repo, as the generation of the Windows VM is a very long process.

File: environments/demo.yml [hunk 5]
```diff
@@ -14,7 +14,6 @@ servers:
 ansible_config:
   extra_vars:
     vagrant: true
-    vagrant_share: false
```
Comment (guillaumededrie) [inline environments/demo.yml]: Good catch! However, I think we should not put this in this commit, but in a new one, as it's not really related to the Windows provisioning.

File: ansible.cfg (hunk 3 above)
Comment (kamino) [inline ansible.cfg]: Yep ! I'll do that definitely (artefact also)

File: README.rst (hunk 4 above)
Comment (kamino) [inline README.rst]: I am not convinced that we can distribute installation of Windows, even if it is an evaluation mode. I would not recommend that.

Comment (kamino) [inline README.rst]: I will do a separate fix. Thanks for the feedback.

File: environments/demo.yml (hunk 5 above)
Comment (kamino) [inline environments/demo.yml]: I will do a separate fix. Thanks for the feedback.

**Comment by guillaumededrie** [comment]
I have been thinking to some improvements for the directory layout.

https://github.com/quarkslab/irma-ansible/tree/1142-ansible-windows-provisioning/roles/quarkslab.irma_deployment_probe/tasks:
 - merge `main_linux.yml` with `linux_probe_deployment.yml` as there is an include without any constraint.
 - merge `main_windows.yml` with `windows_probe_deployment.yml` as there is an include without any constraint.

https://github.com/quarkslab/irma-ansible/tree/1142-ansible-windows-provisioning/roles/quarkslab.irma_provisioning_probe/tasks:
 - split the main into two main as the deployment, and regroup in the windows version : 
    - `windows_install_packages.yml`
    - `windows_configuration.yml`
    - the `Create IRMA application directories` task
    - `pip.yml`
 - Export the `choco*.yml` file to another role (`quarkslab.chocolatey`) it'll be more readable i think, and more reusable for other.

What's do you think about that?

**Comment by kamino** [comment]
I do not want to merge ``windows_install_packages.yml``, ``windows_configuration.yml``. I want to keep them separate and have it called from another file (simply to keep things segmented)
For the ``choco*.yml``, actually I prefer leaving it there as it will be replaced in the future by chocolatey ansible built-in module.

File: README.rst (hunk 4 above)
Comment (guillaumededrie) [inline README.rst]: Yes, I did forget it!

**Comment by kamino** [comment]
Actually, after reviewing windows/linux_probe_deployment, I prefere to keep it that way to separate clearly tasks. File names should be more significant here, but it deserve a separate commit (in any cases, changes does not impact only windows provisioning)

File: Vagrantfile [hunk 6]
```diff
@@ -12,9 +12,10 @@ ansible_config = configuration['ansible_config'] || false
 Vagrant.require_version ">= 1.5.0"
 
 Vagrant.configure("2") do |config|
-  config.ssh.forward_agent = true
+
+  machine.ssh.forward_agent = true
```
Comment (guillaumededrie) [inline Vagrantfile]: It can work as machine is not defined. If pretty sure it's `config` there.

File: Vagrantfile [hunk 7]
```diff
@@ -12,9 +12,10 @@ ansible_config = configuration['ansible_config'] || false
...
   # see https://twitter.com/mitchellh/status/525704126647128064
-  config.ssh.insert_key = false
+  machine.ssh.insert_key = false
```
Comment (guillaumededrie) [inline Vagrantfile]: See https://github.com/quarkslab/irma-ansible/pull/89/files#r26868649.

**Comment by guillaumededrie** [comment]
Ok. Maybe, to help newbie to easily develop new provisioning/deployment tasks, we can add two subdirectory, one related to linux the other to Windows, with Yaml files in there?

**Comment by guillaumededrie** [comment]
Ok, good to go. Need only a squash between the 4 first commit and that's good (don't squash the `removing type in environment/allione_prod.yml`)

**Comment by kamino** [comment]
commit https://github.com/quarkslab/irma-ansible/commit/83ae4d554b217bcb592979ea8bc4a7d1bba64794 fixes #1276

**Pull request closed without merging**